### PR TITLE
FIX-BESTPRACTICES: oms-all-deploy-01

### PR DESCRIPTION
**Pull request by badri-spektra**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

**Comment by badri-spektra** [comment]
We tried to upgrade API versions for all resource types, except below resource types remaining we were able to upgrade them

##[error]        Api versions must be the latest or under 2 years old (730 days) - API version 2019-06-01 of Microsoft.Automation/automationAccounts/credentials is 773 days old
        Valid Api Versions:
        2021-04-01

##[error]        Api versions must be the latest or under 2 years old (730 days) - API version 2019-06-01 of Microsoft.Automation/automationAccounts/Modules is 773 days old
        Valid Api Versions:
        2021-04-01

##[error]        Api versions must be the latest or under 2 years old (730 days) - API version 2019-06-01 of Microsoft.Automation/automationAccounts/runbooks is 773 days old
        Valid Api Versions:
        2020-01-13-preview

##[error]        Microsoft.OperationalInsights/workspaces/Microsoft.OperationalInsights/workspaces/savedSearches uses a preview version ( 2017-03-15-preview ) and there are more recent versions available.
        Valid Api Versions:
        2021-06-01

##[error]        Api versions must be the latest or under 2 years old (730 days) - API version 2015-11-01-preview of Microsoft.OperationalInsights/workspaces/datasources is 2081 days old
        Valid Api Versions:
        2020-08-01


When we try to upgrade above resource types we getting error below which we don't understand

"error": { 
 "code": "BadRequest", 
 "message": "Failed to start operation. Verify input and try operation again.", 
 "details": [ 
 { 
 "code": "FailedToStartOperation", 
 "message": "Failed to start operation. Verify input and try operation again.", 
 "possibleCauses": "Invalid parameters were specified.", 
 "recommendedAction": "Verify the input and try again.", 
 "clientRequestId": "717feaeb-0d20-44e4-84e4-6bf39dc303fb", 
 "activityId": "a84b8fac-303e-466e-97e3-a3ba2630fe96" 
 } 
 ] 
 }

File: demos/oms-all-deploy/nestedtemplates/omsAutomation.json [hunk 1]
```diff
@@ -54,54 +53,21 @@
...
+    },
+    "psModules": {
+      "type": "object",
+      "metadata": {
+        "description": "URIs of the PS modules"
+      }
     }
   },
   "variables": {
-    "psModules": {
```
Comment (bmoore-msft) [inline demos/oms-all-deploy/nestedtemplates/omsAutomation.json:60]: revert this change - don't worry about the BP warning

File: demos/oms-all-deploy/azuredeploy.parameters.json [hunk 2]
```diff
@@ -3,19 +3,77 @@
...
-            "value": "GEN-UNIQUE-30"
+            "value": "GEN-UNIQUE-10"
         },
         "azureAdminPwd": {
             "value": "GEN-PASSWORD"
         },
         "azureAdmin": {
             "value": "azureadmin"
+        },
+        "psModules": {
```
Comment (bmoore-msft) [inline demos/oms-all-deploy/azuredeploy.parameters.json:20]: revert this change

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: demos/oms-all-deploy/nestedtemplates/omsAutomation.json (hunk 1 above)
Comment (badri-spektra) [reply demos/oms-all-deploy/nestedtemplates/omsAutomation.json]: reverted

File: demos/oms-all-deploy/azuredeploy.parameters.json (hunk 2 above)
Comment (badri-spektra) [reply demos/oms-all-deploy/azuredeploy.parameters.json]: reverted
